src/adapters.ts

```diff
@@ -126,10 +145,12 @@ export function AdapterL1<TBase extends Constructor<TxSender>>(Base: TBase) {
         async l2TokenAddress(token: Address): Promise<string> {
             if (token == ETH_ADDRESS) {
                 return ETH_ADDRESS;
+            } else if (token.toLowerCase() == (await this.baseTokenAddress())) {
```

Comment:
I think if someone want to deposit ETH, but the base token is not ETH then it should not be allowed
(only WETH is supported)

In that case we may just return zero address or return an exception?